App/Controllers/LoginController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+
+namespace App\Controllers;
+
+use Core\View;
+use App\Models\User;
+use App\Models\Permission;
+
+class LoginController {
+
+    public function index()
+    {
+        $users = User::getDB()->readAll(User::getTableName(),User::class);
+        $userExists = false;
+        if (count($users) > 0)
+        {
+            $userExists = true;
+        }
+        View::render("login.view.php",["userExists" => $userExists]);
+    }
+
+    public function login()
+    {
+      if(session_status() !== PHP_SESSION_ACTIVE)
+          session_start();
+
+      if(count($_POST)>0)
+      {
+         $sqlParams['user_name']=$_POST["user_name"];
```

Comment:
Create a Validator here which can check for required parameters and see if user submitted user_name or password. Always remember to check if the key exists in array as PHP will complain with an undefined offset.

On PHP7 you can do something like `$_POST["user_name"] ?? "";`

On <PHP7 you can do it like `isset($_POST["user_name"]) ? $_POST["user_name"] : "";"

Reply:
Where should I implement this logic? Inside the LoginController or should I implement a separate class that validates the login form?

Reply:
A separate class under App\Validators which will verify based on form.